src/lib/settings/SettingsFolder.js

```diff
@@ -120,17 +118,17 @@ class SettingsFolder extends Map {
  	 * Resolves paths into their full objects or values depending on the current set value
  	 * @since 0.5.0
  	 * @param  {...string} paths The paths to resolve
- 	 * @returns {*}
+ 	 * @returns {Promise<any[]>}
```

Comment:
using `Promise` in jsdoc when the method is async is redunant due to docsgen (the tool that builds the docs json file) does this for us. but since we're going to remove async from the function this can stay